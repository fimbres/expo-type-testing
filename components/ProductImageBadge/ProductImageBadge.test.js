import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react-native';

import { ProductImageBadge } from '.';

describe('<ProductImageBadge />', () => {
    it('render image correctly', () => {
        render(<ProductImageBadge productImageUri='https://loremflickr.com/640/480/people' />);
        const image = screen.getByTestId('product-image');
        expect(image).toBeDefined();
    });

    it('render correctly', () => {
        const tree = renderer
        .create(
            <ProductImageBadge productImageUri='https://loremflickr.com/640/480/people' />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
