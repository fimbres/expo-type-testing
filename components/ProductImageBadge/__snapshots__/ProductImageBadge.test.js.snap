// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductImageBadge /> render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "aspectRatio": 1,
      "backgroundColor": "#FFFFFF",
      "borderRadius": 10,
      "elevation": 3,
      "justifyContent": "center",
      "marginTop": 19,
      "shadowColor": "#020202",
      "shadowOffset": Object {
        "height": 6,
        "width": 2,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 8,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://loremflickr.com/640/480/people",
      }
    }
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
    testID="product-image"
  />
</View>
`;
